### PR TITLE
Integration test for hotrod on k8s

### PR DESCRIPTION
**Pull request by chahatsagarmain**
## Which problem is this PR solving?
- #6086 

## Description of the changes
- 

## How was this change tested?
- 

## Checklist
- [ ] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [ ] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

**Review by yurishkuro — COMMENTED** [review]
please avoid unnecessary changes. For example, why did the flag parsing moved from the top of the file (where it is customary located in all our scripts) into the main()? If you wanted to wrap it in a function, you could do it without physically moving the code and creating a disassociated code diff.

File: scripts/build-hotrod-image.sh [hunk 1]
```diff
@@ -5,41 +5,22 @@
 
 set -euf -o pipefail
 
+JAEGER_QUERY_URL="http://localhost:16686"
+EXPECTED_SPANS=35
+MAX_RETRIES=30
+SLEEP_INTERVAL=3
```
Comment (yurishkuro) [inline scripts/build-hotrod-image.sh:11]: nit: upper case for env vars, lower case for local vars

File: scripts/build-hotrod-image.sh [hunk 2]
```diff
@@ -5,41 +5,22 @@
...
 print_help() {
-  echo "Usage: $0 [-h] [-l] [-o] [-p platforms]"
+  echo "Usage: $0 [-h] [-l] [-o] [-p platforms] [-d | -k]"
   echo "-h: Print help"
   echo "-l: Enable local-only mode that only pushes images to local registry"
-  echo "-o: overwrite image in the target remote repository even if the semver tag already exists"
+  echo "-o: Overwrite image in the target remote repository even if the semver tag already exists"
   echo "-p: Comma-separated list of platforms to build for (default: all supported)"
+  echo "-d: Run Docker tests only"
+  echo "-k: Run Kubernetes tests only"
```
Comment (yurishkuro) [inline scripts/build-hotrod-image.sh:20]: I'd rather have `-r docker | k8s` with `docker` being default

**Comment by chahatsagarmain** [comment]
@yurishkuro i have made some changes please have a look

**Pull request closed without merging**